filebeat/channel/runner.go

```diff
@@ -134,6 +137,14 @@ func newCommonConfigEditor(
 	}
 
 	return func(clientCfg beat.ClientConfig) (beat.ClientConfig, error) {
+		if config.Dataset != "" {
+			index, err := fmtstr.CompileEvent(fmt.Sprintf("logs-%s-default", config.Dataset))
```

Comment:
As this value is written into the index, I assume also variables could be used like described in https://www.elastic.co/guide/en/beats/filebeat/current/elasticsearch-output.html#index-option-es